### ISSUE TITLE
When changing networks on Android I lose connectivity

### ISSUE THREAD

**Issue by dzfranklin** [tailscale/tailscale#8283]
### What is the issue?

When I disconnect from wifi my phone often loses connectivity until I turn off tailscale. 

### Steps to reproduce

- Connect phone to WiFi
- Turn on tailscale
- Turn off wifi
- Observe
    - Phone indicates it has cell service
    - ping goo.gl: unknown host goo.gl
    - browse example.com in Firefox: address not found
    - browse example.com in Chrome: ERR_NETWORK_CHANGED
- Toggle airplane mode on and off or toggle tailscale off and then back on
- Observe: ping works and websites load

These steps don't consistently reproduce the issue but this happens to me multiple times a week. 

### Are there any recent changes that introduced the issue?

No

### OS

Android

### OS version

Android 11 Build RZKS31.Q3-45-16-8-19 motorola one 5G ace

### Tailscale version

1.40.0

### Other software

None

### Bug report

BUG-39c329541a78618770972e91d86ffdad71a602ed5c64aaa7d9d0f8753d574a35-20230606090946Z-10e7c0b061e79d73

**Comment by jglamine** [comment]
I'm seeing the same problem - see https://github.com/tailscale/tailscale/issues/8312

**Comment by j0hncc** [comment]
I'm seeing this problem too.  New user and not very network savvy so idk if this helps but
BUG-6a57b76247995fe7ba2d7204c2777ab190cfbeb1fc64548c439ff07aac3a5016-20230626015657Z-3d7970f838651177

Samsung A53

**Comment by kelivel** [comment]
Hi folks, I'm investigating these reports and working on reproducing the same problem with a colleague. So far, all of the bugreports I'm looking at are showing DNS resolution problems. Can you reproduce the same behavior after [overriding local DNS](https://login.tailscale.com/admin/dns) on the tailnet? If you can, send a fresh bugreport after resolution fails and I'll take a closer look. Thanks!

**Comment by danielzfranklin** [comment]
@kelivel I can't reproduce the issue after overriding local DNS. Thanks!

**Comment by robsel-0** [comment]
I have this problem with the Tailscale app version 1.90.4.

Network tests performed using an app called "Net Analyzer":
 
* Pinging google.com does not work.
* Pinging IP addresses works.
* DNS resolving using 100.100.100.100 as DNS does not work.
* DNS resolving using 1.1.1.1 works.

**Edit:** Like @jsmucr below, I can also fix my network by turning Tailscale on and off again.
To Tailscale: Please implement a network watchdog and make this resolve automatically.

**Comment by jsmucr** [comment]
Quite recently I started to have this problem as well. Turning Tailscale off and on again seems to help.

**Comment by pavdi7** [comment]
I have same problem on my android last 2 weeks. I have to turn off and on tailscale evertime, when I go between wifi and cellular connection. Tailscale (1.90.4) in phone displays the message "DNS unavailable" almost all the time, even in situations where the connection seems to be fine.

bug report: BUG-5dc9a44f2386b61007e6e030c10756f0b6c35ea521028bfd5bf3100bff8e9e33-20251121094442Z-9d6d950b5e53aa5f

edit: I setup global nameserver in admin console (quad9 and cloudflare) and now it seems ok. Android no more displays message "DNS unavailable" and connection dont fall down.

**Comment by MrVideo** [comment]
Hello, I too get the same issue on version 1.102.3-t9329c3677-gaea8f60c0.

DNS connectivity drops randomly and this only happens on Android: I have several iOS devices, a Mac and a Windows PC which never have the same issue with DNS. My global tailnet nameserver is my home server with Pi-Hole, and "Override DNS servers" is on, but it does not fix my issue.

One thing that I've noticed is that it doesn't seem to follow a pattern: manually toggling Wi-Fi off and on again does not break DNS, but often I have no resolution when off Wi-Fi. Restarting Tailscale fixes it.

I have disabled Private DNS in my devices settings and it seemed to help, but the problem resurfaced anyway.
